src/main/java/mate/academy/bookstore/controller/BookController.java

```diff
@@ -39,19 +40,22 @@ public BookDto findById(@PathVariable Long id) {
         return bookService.finById(id);
     }
 
+    @PreAuthorize("hasAuthority('ADMIN')")
     @PostMapping
     @ResponseStatus(HttpStatus.CREATED)
     @Operation(summary = "Create a new book", description = "Create a new book")
     public BookDto save(@RequestBody @Valid CreateBookRequestDto requestDto) {
         return bookService.save(requestDto);
     }
 
+    @PreAuthorize("hasAuthority('ADMIN')")
     @ResponseStatus(HttpStatus.NO_CONTENT)
     @DeleteMapping("/{id}")
     public void delete(@PathVariable Long id) {
         bookService.deleteById(id);
     }
 
+    @PreAuthorize("hasAuthority('ADMIN')")
```

Comment:
Is it needed to add @PreAuthorize to other methods? Also, think about which endpoint could be called by USER and ADMIN at the same time?